### ISSUE TITLE
portable: support attaching portables to user instance

### ISSUE THREAD

**Issue by hundeboll** [systemd/systemd#10782]
**Is your feature request related to a problem? Please describe.**
I would like to be able to attach portables as a non-root user. My use case is automated testing of an application that interacts with systemd using sd_bus. Since it expects certain unit files to be present, a systemd portable would be perfect to "install" those.

**Describe the solution you'd like**
In all its simplicity, I would simply like run `portablectl` with the `--user` argument:
```
portablectl --user --runtime test/portable app
```

**Describe alternatives you've considered**
An alternative (and maybe my current temporary) solution is to install a couple of units into `$XDG_RUNTIME_DIR/systemd/user/`

**Comment by poettering** [comment]
This is really hard to do, as the feature requires fs namespaces to work and we can't really have that as non-root so easily.

**Comment by kirbyfan64** [comment]
What about an suid sandbox like bubblewrap?

**Comment by poettering** [comment]
We have unpriv RootImage= finally. So it's just a matter now of actually teaching portabled and portablectl two different image paths and runtime scopes.

**Comment by calebccff** [comment]
running into this in 2024! we're running systemd on musl and would love to run some proprietary glibc binaries as user. portablectl is oviously the perfect solution here.

We may need some system services too, so simply having a system-wide portable image which has system and (global) user services in would be totally sensible for us.

is there any interest in revisiting this?

**Comment by poettering** [comment]
yeah, adding per-user portabled support sounds great. i think all major stumbling blocks are gone by now.

**Comment by daandemeyer** [comment]
Implemented by https://github.com/systemd/systemd/pull/40091
